app/services/pipeline/storage/rubric_service.rb

```diff
@@ -7,18 +7,67 @@ class RubricService
       def self.call(context:)
         Rails.logger.info("Storing rubric for assignment: #{context.assignment.id}")
 
-        # This would contain actual persistence logic in a real implementation
-        # Following PORO principles - keep it simple and focused
-        context.rubric = create_rubric(context)
+        context.rubric = create_rubric_with_criteria_and_levels(context)
         context
       end
 
       private
 
+      def self.create_rubric_with_criteria_and_levels(context)
+        rubric = nil
+
+        ActiveRecord::Base.transaction do
+          rubric = create_rubric(context)
+          create_criteria_with_levels(rubric, context.parsed_response)
+        end
+
+        rubric
+      rescue => e
+        Rails.logger.error("Failed to create rubric: #{e.message}")
+        raise
+      end
+
       def self.create_rubric(context)
-        # Simple implementation that would be expanded in a real implementation
-        # Just returning a mock rubric object for now
-        Rubric.new(assignment: context.assignment)
+        Rubric.create!(assignment: context.assignment)
```

Comment:
We already create the rubric in app/services/assignments/creation_service.rb - we should find the associated rubric instead - `context.rubric`